File: VB/RepeatDetailBandExample/XtraReport1.vb
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing

Namespace RepeatDetailBandExample
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport

		Private currentDetail As Integer = 1
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrLabel5_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrLabel5.BeforePrint
			DirectCast(sender, XRLabel).Text &= currentDetail.ToString()
			DirectCast(sender, XRLabel).Text &= "/"
			DirectCast(sender, XRLabel).Text &= DetailReport.ReportPrintOptions.DetailCountOnEmptyDataSource.ToString()
			currentDetail += 1
		End Sub

		Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles Detail.BeforePrint
			currentDetail = 1
			DetailReport.ReportPrintOptions.DetailCountOnEmptyDataSource = Convert.ToInt32(GetCurrentColumnValue("CategoryID"))
		End Sub
	End Class
End Namespace

File: CS/RepeatDetailBandExample/XtraReport1.cs
using DevExpress.XtraReports.UI;
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;

namespace RepeatDetailBandExample
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        private int currentDetail = 1;
        public XtraReport1()
        {
            InitializeComponent();
        }

        private void xrLabel5_BeforePrint(object sender, System.ComponentModel.CancelEventArgs e)
        {
            ((XRLabel)sender).Text += currentDetail.ToString();
            ((XRLabel)sender).Text += @"/";
            ((XRLabel)sender).Text += DetailReport.ReportPrintOptions.DetailCountOnEmptyDataSource.ToString();
            currentDetail++;
        }

        private void Detail_BeforePrint(object sender, System.ComponentModel.CancelEventArgs e)
        {
            currentDetail = 1;
            DetailReport.ReportPrintOptions.DetailCountOnEmptyDataSource = 
                Convert.ToInt32(GetCurrentColumnValue("CategoryID"));
        }
    }
}
